### PR TITLE
add `evalTerm`

### PR DESCRIPTION
**Pull request by SeungheonOh**
Sorry for making another PR. I made a mistake in the other one. This one should be good to go. 

`evalTerm` evaluates `ClosedTerm` and put it back to `ClosedTerm` using `RCompiled`.

### CODE REVIEW

**Review by L-as — CHANGES_REQUESTED** [review]
Thanks!

File: Plutarch/Internal.hs [hunk 1]
```diff
@@ -471,3 +474,19 @@ compile config t = case asClosedRawTerm t of
 
 hashTerm :: Config -> ClosedTerm a -> Either Text Dig
 hashTerm config t = hashRawTerm . getTerm <$> runReaderT (runTermMonad $ asRawTerm t 0) config
+
+-- | Compile and evaluate term.
+evalTerm ::
```
[the remark this answers is not in the document]
Comment (SeungheonOh) [reply Plutarch/Internal.hs]: Thank you. I didn't know how to do this.

**Review by L-as — APPROVED** [review]
LGTM

**Comment by L-as** [comment]
Thanks!

**Comment by L-as** [comment]
(waiting for CI)

**Comment by SeungheonOh** [comment]
Good to go?

**Pull request merged**